### PR TITLE
Align manager_ref with unique indexes

### PR DESCRIPTION
**Pull request by Ladas**
Align manager_ref with unique indexes PR https://github.com/ManageIQ/manageiq-schema/pull/19

### CODE REVIEW

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb [hunk 1]
```diff
@@ -244,7 +243,7 @@ def initialize_inventory_collections
           :model_class => ContainerServicePortConfig,
           :parent      => manager,
           :association => :container_service_port_configs,
-          :manager_ref => [:ems_ref, :protocol] # TODO(lsmola) make protocol part of the ems_ref?
+          :manager_ref => [:container_service, :name]
```
Comment (Ladas) [inline app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: @cben is this a valid replacement? We do unique index like this

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb [hunk 2]
```diff
@@ -317,8 +316,9 @@ def initialize_inventory_collections
         shared_options.merge(
           :model_class    => PersistentVolume,
           :parent         => manager,
-          :builder_params => {:parent => manager},
+          :builder_params => {:parent_id => manager.id, :parent_type => manager.class.base_class},
```
Comment (Ladas) [inline app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: @cben active record object that is part of the manager_ref can act weird now

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb [hunk 3]
```diff
@@ -195,8 +195,7 @@ def initialize_inventory_collections
           :model_class => ContainerEnvVar,
           :parent      => manager,
           :association => :container_env_vars,
-          # TODO: old save matches on all :name, :value, :field_path - does this matter?
-          :manager_ref => [:container, :name],
+          :manager_ref => [:container, :name, :value, :field_path],
```
Comment (Ladas) [inline app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: @cben we have unique index like this, should we reduce the list?

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb (hunk 1 above)
Comment (Ladas) [inline app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: hm, no this is not enough ```DETAIL:  Key (container_service_id, name)=(174, dns) already exists.```

Comment (Ladas) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: wait,the graph refresh should not allow this, I am doing something wrong

Comment (Ladas) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: disregard the other comments, this passed, I had to update old saving code :-)

Comment (cben) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: you mean "name is not unique" [comment](https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/168#issuecomment-356729210)?  it is unique within a service?

Comment (Ladas) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: no disregard that, it was too late :-) I was missing the PR for old refresh. Now this passes all specs, ao it should be probably good :-)

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb (hunk 3 above)
Comment (cben) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: hmm.  my instinct is to get rid of value and field_path (which is an [alternative to value](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-container-fields-as-values-for-environment-variables)).
But I need to verify you can't define 2 vars with same name.

Comment (Ladas) [reply app/models/manageiq/providers/kubernetes/container_manager/inventory_collections.rb]: Ok, let me know then.

**Pull request closed without merging**